test/benchmark-common-tasks/main.cpp

```diff
@@ -593,6 +596,95 @@ struct BenchmarkQueryLongString : BenchmarkWithStrings {
     }
 };
 
+struct BenchmarkQueryInsensitiveString : BenchmarkWithStringsTable {
+    const char* name() const
+    {
+        return "QueryInsensitiveString";
+    }
+
+    std::string gen_random_case_string(size_t length)
+    {
+        std::stringstream ss;
+        for (size_t c = 0; c < length; ++c) {
+            bool lowercase = (rand() % 2) == 0;
+            // choose characters from a-z or A-Z
+            ss << (rand() % 26) + (lowercase ? 97 : 65);
+        }
+        return ss.str();
+    }
+
+    std::string shuffle_case(std::string input)
+    {
+        for (size_t i = 0; i < input.size(); ++i) {
+            bool upper = (rand() % 2) == 0;
+            int c = input[i];
```

Comment:
This benchmark is only meant to show that there is an overall improvement in performance in case insensitive string queries after the string index work. This should be clear in the graphs; it is on average 26 times faster now even with these random characters. I agree that we should make a benchmark with "real" strings at some point too.

> `str[i] ^= change_case ? 0x20 : 0;`

Nice creative idea! I'm gonna steal it 😄